### PR TITLE
=htc #19814 fix port handling in ConnectHttp

### PR DESCRIPTION
**Pull request by ktoso**
Resolves https://github.com/akka/akka/issues/19814
Will have another look if I didn't miss a case tomorrow morning

### CODE REVIEW

File: akka-http-core/src/main/scala/akka/http/javadsl/ConnectHttp.scala [hunk 1]
```diff
@@ -66,13 +67,13 @@ object ConnectHttp {
   @throws(classOf[IllegalArgumentException])
   def toHostHttps(host: String, port: Int): ConnectWithHttps = {
     require(port > 0, "port must be > 0")
-    val start = if (host.startsWith("https://")) host else s"https://$host"
+    val start = if (host.startsWith("http://") || host.startsWith("https://")) host else s"https://$host"
     toHostHttps(Uri.create(s"$start").port(port))
```
Comment (johanandren) [inline akka-http-core/src/main/scala/akka/http/javadsl/ConnectHttp.scala]: Why the `s"$start"` here, can't we just pass `start`?

File: akka-http-core/src/main/scala/akka/http/javadsl/ConnectHttp.scala [hunk 2]
```diff
@@ -66,13 +67,13 @@ object ConnectHttp {
   @throws(classOf[IllegalArgumentException])
   def toHostHttps(host: String, port: Int): ConnectWithHttps = {
     require(port > 0, "port must be > 0")
-    val start = if (host.startsWith("https://")) host else s"https://$host"
+    val start = if (host.startsWith("http://") || host.startsWith("https://")) host else s"https://$host"
```
Comment (johanandren) [inline akka-http-core/src/main/scala/akka/http/javadsl/ConnectHttp.scala]: Why does `toHost(Uri)` need lowercasing of the scheme, but this not?

Comment (johanandren) [inline akka-http-core/src/main/scala/akka/http/javadsl/ConnectHttp.scala]: also, a lot of repeated identical string comparisons here, couldn't those be extracted to one method instead?

Comment (johanandren) [inline akka-http-core/src/main/scala/akka/http/javadsl/ConnectHttp.scala]: or well, not exactly identical but all comparing strings to "http" and "https" in different ways

Comment (ktoso) [inline akka-http-core/src/main/scala/akka/http/javadsl/ConnectHttp.scala]: I'll try to DRY it up a bit

File: akka-http-core/src/test/scala/akka/http/javadsl/ConnectHttpSpec.scala [hunk 3]
```diff
@@ -0,0 +1,75 @@
...
+      connect.port should ===(443)
+    }
+    "connect toHostHttps HTTPS:8080 when https prefix and port given in host" in {
+      val connect = ConnectHttp.toHostHttps("https://127.0.0.1:8080")
+      connect.isHttps should ===(true)
+      connect.connectionContext.isPresent should equal(false)
+      connect.host should ===("127.0.0.1")
+      connect.port should ===(8080)
+    }
+    "connect toHostHttps HTTPS:8080 when https prefix and port given" in {
```
Comment (drewhk) [inline akka-http-core/src/test/scala/akka/http/javadsl/ConnectHttpSpec.scala]: "9999" not "8080"

**Comment by drewhk** [comment]
LGTM

**Comment by ktoso** [comment]
Failure was unrelated (ssh issue).

File: akka-http-core/src/test/scala/akka/http/javadsl/ConnectHttpSpec.scala (hunk 3 above)
Comment (ktoso) [inline akka-http-core/src/test/scala/akka/http/javadsl/ConnectHttpSpec.scala]: thanks

**Pull request merged**